### ISSUE TITLE
Benchmark different parallelization strategy for scoring

### ISSUE THREAD

**Issue by hombit** [snad-space/coniferest#117]
Currently, we parallelize scoring over data sample evaluating trees subsequently. It would be interesting to check what if we parallelize over trees, especially for the case of large tree number.

**Comment by balodja** [comment]
We know that the large sample size is a thing. Do we know any situations when we have to evaluate scores over millions of trees?

**Comment by hombit** [comment]
@balodja Do I understand correctly that we are talking about memory caching here? For me it looks like both caching of the whole data set and caching of the whole forest are unlikely, and my intuition is not enough to say what is faster

**Comment by balodja** [comment]
@hombit , yes, we are talking about memory caching. My hope is that when you parallelize over the data the whole selectors array (less than 2MB) may fit part the L3 cache and CPU only have to load data points from memory.
